### PR TITLE
Create routes for index and show for book

### PR DESCRIPTION
**Pull request by lucasVoboril**
## Summary

The model book should have index and show routes. To do this it's necessary users had been  authenticated.
For this requirement to be accepted:
- [x] Use Active Model Serializers
- [X] Paginated end points of index using wor-paginate 

## Screenshots

pr photo

## Trello Card

https://trello.com/c/hb4MjCDP

### CODE REVIEW

File: app/controllers/book_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class BookController < ApplicationController
+  include Wor::Paginate
+
+  def index
+    @books = Book.filter(params.require(:book).permit(%i[genre title author]))
```
Comment (Martum) [inline app/controllers/book_controller.rb]: For an index endpoint we usually want to get all the records if no filter is used. The `.require(:book)` will raise an exception if no filter is passed! Do you have any idea how can we handle this to return the full collection if no filter is used??

Also, variables starting with `@` are attached to the instance, in this case we just need a local variable.

File: app/controllers/book_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    render_paginated @books, each_serializer: BookSerializer
+  end
+
+  def show
+    @book = Book.filter(params.slice(id))
```
Comment (Martum) [inline app/controllers/book_controller.rb]: You are assigning the result to a variable but you aren't using it 😱Maybe you are missing a render here?? 🤔  
If you want to find an instances by its ID, you can use [find](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). For example, `Book.find(params[:id])`

File: app/controllers/book_controller.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class BookController < ApplicationController
```
Comment (Martum) [inline app/controllers/book_controller.rb]: A very good pattern is to have two different base controllers, one for API's endpoints (`ApiControler`) and one for views which render content (`ApplicationController`).
Feel free to create a new `ApiController` with the exact same content of `ApplicationController`, and change the inheritance here to use `ApiController`!

File: app/lib/filterable.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module Filterable
```
Comment (Martum) [inline app/lib/filterable.rb]: The default location for models' concerns is `app/models/concerns`, please move this file to there!

File: config/routes.rb [hunk 5]
```diff
@@ -1,3 +1,9 @@
 Rails.application.routes.draw do
   mount_devise_token_auth_for 'User', at: 'auth'
+
+  authenticate :user do
```
Comment (Martum) [inline config/routes.rb:4]: Didn't know this feature! The usual way is declaring a `before_action :authenticate_request!` in the controller, but we can keep this!

**Review by Martum — CHANGES_REQUESTED** [review]

File: .rubocop.yml [hunk 6]
```diff
@@ -16,7 +16,7 @@ Documentation:
   Enabled: false
 
 LineLength:
-  Max: 99
+  Max: 110
```
Comment (Martum) [inline .rubocop.yml]: Change this isn't a good practice! If you have a line which is longer than 99 characters, you can disable this check to that line. Let's keep this global configuration as it was.

File: config/routes.rb [hunk 7]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
   mount_devise_token_auth_for 'User', at: 'auth'
+
+  get '/books' => 'book#index', as: 'books'
```
Comment (Martum) [inline config/routes.rb]: I would use the default route generator for resources, in this case is
```ruby
resources :books, only: [:index, :show]
```

File: app/controllers/api_controller.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+class ApiController < ActionController::Base
+  include DeviseTokenAuth::Concerns::SetUserByToken
+  protect_from_forgery with: :exception
```
Comment (Martum) [inline app/controllers/api_controller.rb]: In APIs always use `with: :null_session`

**Review by Martum — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 7 above)
Comment (lucasVoboril) [reply config/routes.rb]: I considered doing it this way. But it does not work :sweat_smile:

File: config/initializers/ams.rb [hunk 9]
```diff
@@ -0,0 +1 @@
+ActiveModelSerializers.config.adapter = :json_api
```
Comment (Martum) [inline config/initializers/ams.rb:1]: We use `:json` as a serializer. JSONApi is a standard for APIs, but we don't use it.

**Review by Martum — APPROVED** [review]

**Pull request merged**